x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchResponseTests.java

```diff
@@ -217,5 +217,36 @@ public static EqlSearchResponse createRandomInstance(TotalHits totalHits, XConte
                 return null;
         }
     }
-    
+
+    @Override
+    protected EqlSearchResponse mutateInstanceForVersion(EqlSearchResponse instance, Version version) {
+        List<Event> mutatedEvents = mutateEvents(instance.hits().events(), version);
+
+        List<Sequence> sequences = instance.hits().sequences();
+        List<Sequence> mutatedSequences = null;
+        if (sequences != null) {
+            mutatedSequences = new ArrayList<>(sequences.size());
+            for(Sequence s : sequences) {
+                mutatedSequences.add(new Sequence(s.joinKeys(), mutateEvents(s.events(), version)));
+            }
+        }
+        
+        return new EqlSearchResponse(new EqlSearchResponse.Hits(mutatedEvents, mutatedSequences, instance.hits().totalHits()),
+            instance.took(), instance.isTimeout(), instance.id(), instance.isRunning(), instance.isPartial());
+    }
+
+    private List<Event> mutateEvents(List<Event> original, Version version) {
+        if (original == null || original.isEmpty()) {
+            return original;
+        }
+        List<Event> mutatedEvents = new ArrayList<>(original.size());
+        for(Event e : original) {
+            if (version.onOrAfter(Version.V_7_13_0)) {
+                mutatedEvents.add(new Event(e.index(), e.id(), e.source(), e.fetchFields()));
+            } else {
+                mutatedEvents.add(new Event(e.index(), e.id(), e.source(), null));
+            }
```

Comment:
nit: optionally, you could add one line (or two) only, if you condition just last parameter's value by the version (like done in the Request test).